src/Castle.Core/DynamicProxy/Contributors/ClassProxyTargetContributor.cs

```diff
@@ -42,6 +42,16 @@ protected override IEnumerable<MembersCollector> GetCollectors()
 			var targetItem = new ClassMembersCollector(targetType) { Logger = Logger };
 			yield return targetItem;
 
+			foreach (var @interface in targetType.GetAllInterfaces())
```

Comment:
I have been wondering whether adding a cache for `type.GetAllInterfaces()` might be a good idea. It'll now be called at least twice per class proxy type generation (and IIRC it's already being called twice per interface proxy type generation). Probably not the most urgent optimization, but might be good looking into that at some point.

Reply:
You'd have to confirm the performance impact, as I believe the runtime team has made reflection improvements in more recent releases to better cache specific type information.